compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2495,3 +2499,36 @@ fn get_apple_sdk_root(sdk_name: &str) -> Result<String, String> {
         Err(e) => Err(format!("failed to get {} SDK path: {}", sdk_name, e)),
     }
 }
+
+fn add_gcc_ld_path(
+    cmd: &mut dyn Linker,
+    sess: &Session,
+    crt_objects_fallback: bool,
+    flavor: LinkerFlavor,
+) {
+    if let Some(ld_impl) = sess.opts.debugging_opts.gcc_ld {
+        if let LinkerFlavor::Gcc = flavor {
+            match ld_impl {
+                LdImpl::Lld => {
+                    let tools_path = sess
+                        .host_filesearch(PathKind::All)
+                        .get_tools_search_paths(crt_objects_fallback);
```

Comment:
```suggestion
                        .get_tools_search_paths(false);
```
The newly added linker file is never placed to the `self-contained` directory.